src/lib/storage/index/group_key/group_key_index.cpp

```diff
@@ -14,41 +15,67 @@ size_t GroupKeyIndex::estimate_memory_consumption(ChunkOffset row_count, ChunkOf
 }
 
 GroupKeyIndex::GroupKeyIndex(const std::vector<std::shared_ptr<const BaseSegment>>& segments_to_index)
-    : BaseIndex{get_index_type_of<GroupKeyIndex>()},
-      _indexed_segments(std::dynamic_pointer_cast<const BaseDictionarySegment>(segments_to_index[0])) {
-  Assert(static_cast<bool>(_indexed_segments), "GroupKeyIndex only works with dictionary segments_to_index.");
+    : AbstractIndex{get_index_type_of<GroupKeyIndex>()},
+      _indexed_segment(segments_to_index.empty()  // Empty segment list is illegal
+                           ? nullptr              // but range check needed for accessing the first segment
+                           : std::dynamic_pointer_cast<const BaseDictionarySegment>(segments_to_index[0])) {
+  Assert(static_cast<bool>(_indexed_segment), "GroupKeyIndex only works with dictionary segments_to_index.");
   Assert((segments_to_index.size() == 1), "GroupKeyIndex only works with a single segment.");
 
-  // 1) Initialize the index structures
-  // 1a) Set the index_offset to size of the dictionary + 1 (plus one to mark the ending position)
-  //     and set all offsets to 0
-  _index_offsets = std::vector<size_t>(_indexed_segments->unique_values_count() + 1u, 0u);
-  // 1b) Set the _index_postings to the size of the attribute vector
-  _index_postings = std::vector<ChunkOffset>(_indexed_segments->size());
+  // 1) Initialize the index_offset:
+  //    Set the index_offset to size of the dictionary + 1 (plus one to mark the ending position)
```

Comment:
Not your fault, but "the index_offset"? Isn't it "Create _index_offsets with a size"? Same in the "title" of this block